[ES-34] Add overview information about EthSigner. (#16)
PegaSysEng/doc.ethsigner: docs/index.md

# EthSigner Transaction Signer

## What is Ethsigner?

> EthSigner is an open-source, client-agnostic, Ethereum transaction signer developed under the Apache 2.0 license and written in Java. It adds a layer of security by signing transactions using a private key that can be secured in a variety of cloud providers, or encrypted on a local disk. EthSigner therefore separates private key management from transaction validation. 

Reviewer: Editing comments - I'd rewrite the second sentence to remove 'It' (it's not always clear what it/this refer to without rereading the preceding sentence): 
A layer of security is added by signing transactions using a private key secured in a cloud provider, or encrypted on a local disk.

I think 'therefore' could be removed from the last sentence

Author: I'll combine the last 2 sentences so that it reads, "EthSigner separates private key management from transaction validation by signing transactions using a private key that can be secured in a variety of cloud providers, or encrypted on a local disk."

> EthSigner is an open-source, client-agnostic, Ethereum transaction signer developed under the Apache 2.0 license and written in Java. It adds a layer of security when operating in Ethereum by signing transactions using a private key that can be secured in a variety of cloud providers, or encrypted on a local disk. EthSigner therefore separates private key management from transaction validation. 

Author: Yep, it's probably redundant, so I'll update this.

## How does it work?

> EthSigner acts as a proxy service by forwarding requests between the client and network. When it receives a transaction it will generate a signature using the stored private key, and forward the transaction and signature to the client to include in the blockchain.

Reviewer: Editing comments - use present tense (ie, don't use _will_) and again may not be clear what _it_ refers to without rereading previous sentence: 

```suggestion
EthSigner acts as a proxy service by forwarding requests to the client. When EthSigner receives a transaction it generates a signature using the stored private key, and forwards the signed transaction to the client.
```

Reviewer: EthSigner forwards requests to the client rather than between the client and the network

> EthSigner acts as a proxy service by forwarding requests between the client and network. When it receives a transaction it will generate a signature using the stored private key, and forward the transaction and signature to the client to include in the blockchain.

Reviewer 2: This is kind hard - people refer to Pantheon as the "Ethereum Client" - but I'm wondering if in this sentence - its meaning the client _application_?

Author: I think Madelines suggestion regarding this paragraph will remove the ambiguity. I'll also be specific and call it an Ethereum client.

!!! note

>     EthSigner currently supports a single private key only. That is, only one private key can be stored in the location specified when starting EthSigner.

Reviewer 2: Reword subtly?
... That is, only one private key's location can be specified when starting EthSigner ...

EthSigner can sign transactions with a key stored in:

> * A V3 keystore file

Reviewer 2: append with "(.stored on a file system accessible by the host)".

* [Hashicorp Vault](Using-EthSigner/Hashicorp.md) 
* [Azure Key Vault](Using-EthSigner/Azure.md). 

> The transaction process when using EthSigner is as follows:

Reviewer: ```suggestion
The transaction process when using EthSigner is:
```

Remove directional language (ie, the _as follows_ is implied and not needed)

Reviewer: I like the diagram :-)
